src/ScalarMessaging.js

```diff
@@ -570,6 +617,16 @@ const onMessage = function(event) {
 
     const roomId = event.data.room_id;
     const userId = event.data.user_id;
+
+    // These APIs don't require roomId
```

Comment:
When a room is supplied, it should be validated, which is what the check at https://github.com/matrix-org/matrix-react-sdk/pull/1672/files#diff-b7b396a3a7ddc61d965d561574b81f63R648 is doing. Having the room ID passed along before this check means that one can grab widgets for any room, even the one the user isn't looking at.

Reply:
Thanks for pointing this out. Hopefully handled in 46f94b3c5aab156b6b6d1406fa7d5aa00fc85089